Determine project type from files in repo

rmunn opened the pull request:
This uses the hgweb URL for browsing the files in the repo's tip to get a list of files, and uses a filename-based heuristic to determine the project type from the list of files in the repo.

Fixes #199.

myieye commented:
@rmunn Regarding OurWord projects, it looks like you want to look for either an `*.otrans` or `*.owp` file in a folder called `.Settings`. I don't quite feel comfortable posting a link, but you saw the one on slack.

hahn-kev commented on backend/LexBoxApi/Controllers/ProjectController.cs:
```diff
@@ -68,6 +69,45 @@ public async Task<ActionResult<int>> AutoUpdateUnknownProjectTypes()
+    {
+        var projects =
+            _lexBoxDbContext.Projects
+            .Where(p => p.Type == ProjectType.Unknown)
```
lets only do this on migrated projects

rmunn commented:
Just rebased this PR on top of current `develop` branch, and added a [commit](https://github.com/sillsdev/languageforge-lexbox/pull/473/commits/370e287702e0d0978bf1e535ea538adce7127f32) that adds a button in the UI to just look up (without updating) a project type. That will allow me to easily test the project code logic locally, then revert that commit before merging the PR.

hahn-kev commented:
@rmunn is this ready for review?

hahn-kev commented:
I think this is good to go so I'm going to merge it in. I added a check for a .Settings folder, but it will only consider it to be OurWord if all the other checks fail.
